tests/publiccloud/pcw_client.pm

```diff
@@ -0,0 +1,37 @@
+use base "consoletest";
+use serial_terminal 'select_serial_terminal';
+use strict;
+use warnings;
+use testapi;
+use mmapi;
+use utils qw(zypper_call);
+use network_utils qw(setup_static_network);
+use lockapi;
+
+sub run {
+    my ($self, $args) = @_;
+    select_serial_terminal;
+
+    record_info('system', script_output('cat /etc/os-release'));
+    setup_static_network(ip => '10.0.2.15/15', gw => '10.0.2.2', iface => 'eth0');
+    assert_script_run('echo "10.0.2.20  microos" >> /etc/hosts');
+    zypper_call('in -y iputils git');
+
+    assert_script_run('mkdir /root/.ssh');
+    assert_script_run('curl -f -v ' . autoinst_url . '/data/slenkins/ssh/id_rsa > /root/.ssh/id_rsa');
+    assert_script_run('chmod 600 /root/.ssh/id_rsa');
+
+    my $children = get_children();
+    my $child_id = (keys %$children)[0];
+    mutex_wait('target_is_ready', $child_id);
+
+    # Testing target is accessible
+    assert_script_run('ping -c 1 microos');
+    assert_script_run('ssh -v -o StrictHostKeyChecking=accept-new -o KexAlgorithms=ecdh-sha2-nistp521 root@microos hostname');
```

Comment:
Interesting solution, but I'd still prefer:
```suggestion
    assert_script_run('ssh-keyscan microos >> ~/.ssh/known_hosts');
    assert_script_run('ssh root@microos hostname');
```

Reply:
For somereason I had to add this option KexAlgorithms=ecdh-sha2-nistp521, else this doesn't work
I tried to add the kex algoritms on sshd_config, but seems is not working
```
KexAlgorithms ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group14-sha1,diffie-hellman-group-exchange-sha256
```

Reply:
That should really not be needed. Please try without it with `ssh -v` and let me know the output. Or we can have a call about it.

Reply:
yes, works, I don't know why this didn't work in the past tests